core/src/main/java/jenkins/util/xml/XMLUtils.java

```diff
@@ -29,6 +43,31 @@
     private final static Logger LOGGER = LogManager.getLogManager().getLogger(XMLUtils.class.getName());
     private final static String DISABLED_PROPERTY_NAME = XMLUtils.class.getName() + ".disableXXEPrevention";
 
+    public static final String FEATURE_HTTP_XML_ORG_SAX_FEATURES_EXTERNAL_GENERAL_ENTITIES = "http://xml.org/sax/features/external-general-entities";
+    public static final String FEATURE_HTTP_XML_ORG_SAX_FEATURES_EXTERNAL_PARAMETER_ENTITIES = "http://xml.org/sax/features/external-parameter-entities";
+
+    private static final DocumentBuilderFactory documentBuilderFactory;
+
+    static {
+        documentBuilderFactory = DocumentBuilderFactory.newInstance();
```

Comment:
see above - needs to be done each time you get a new DBF without caching (I don't think it is worth the overhead of a ThreadLocal - as most likely you are loading a resource from the disk or net which will involve IO anyway)